datafusion/physical-expr/src/expressions/is_not_null.rs

```diff
@@ -73,9 +74,16 @@ impl PhysicalExpr for IsNotNullExpr {
     fn evaluate(&self, batch: &RecordBatch) -> Result<ColumnarValue> {
         let arg = self.arg.evaluate(batch)?;
         match arg {
-            ColumnarValue::Array(array) => Ok(ColumnarValue::Array(Arc::new(
-                compute::is_not_null(array.as_ref())?,
-            ))),
+            ColumnarValue::Array(array) => {
+                let bool_array = if let Some(union_array) =
```

Comment:
I was going to suggest adding using `logical_nulls` here but it seems as if `compute::is_not_null` already does that: https://docs.rs/arrow-arith/52.1.0/src/arrow_arith/boolean.rs.html#345-350

Filed https://github.com/apache/arrow-rs/issues/6017 for the upstream bug

```suggestion
                // workaround https://github.com/apache/arrow-rs/issues/6017
                let bool_array = if let Some(union_array) =
```